### DISCUSSION TITLE
Conditional where clauses

### DISCUSSION THREAD

**Discussion by seanrockster** [ravendb/ravendb#19573] [Q&A] [answered]
Hi

I'm trying to build a query however some of the 'where' conditions are conditional, based on another value, but when I run the query the status report the total value of the collection rather than what should be returned by the query.

Here i'm adding a where clause, the condition is not important, but i'm adding after creating the query. Here the 'stats' report the entire document count of the collection.
```
var query = session
                    .Query<CustomerOrder, OrderSearch>()
                    .Statistics(out QueryStatistics stats);                 
                    
query.Where(x => x.AccountId == "xxxxxxx");
                    
var results = await query.Skip(skip).Take(pageSize).ToListAsync();
```

Is there a way to add additional 'where' clauses after the query is created?

Thanks

**Comment by Lwiel** [comment, marked as answer]
Are you using `Where` method from `RavenQueryableExtensions` or LINQ? I checked the first case and it works fine. Note that you have to import it the following way
```csharp
using Raven.Client.Documents.Linq;
```

You should also assign updated query (with `Where` clause) to a variable
```csharp
query = query.Where(x => x.AccountId == "xxxxxxx");
```

You can also add `Statistics` method after `Where` by using [DocumentQuery](https://ravendb.net/docs/article-page/6.2/csharp/client-api/session/querying/document-query/what-is-document-query), e.g.

```csharp
var query = session.Advanced.DocumentQuery<CustomerOrder, OrderSearch>();
                    
if (shouldAddWhereClause)
    query = query.WhereEquals(x => x.AccountId, "xxxxxxx");
    
var results = query.Skip(skip).Take(take).Statistics(out var statistics).ToList();
```

**Comment by seanrockster** [comment]
Thanks
